docs/preview/02-Features/02-Security/auto-restart-servicebus-messagepump-on-rotated-credentials.md

```diff
@@ -38,23 +38,22 @@ public class Startup
     
         string secretName = hostContext.Configuration["ARCUS_KEYVAULT_CONNECTIONSTRINGSECRETNAME"];
         services.AddServiceBusQueueMessagePump(secretName, options => options.JobId =   
-                .WithServiceBusMessageHandler<OrdersAzureServiceBusMessageHandler, Order>();
-
-         // This extension will be available to you once you installed the package.
-         services.AddAutoRestartServiceBusMessagePumpOnRotatedCredentialsBackgroundJob(
-             jobId: jobId, 
-             subscriptionNamePrefix: "TestSub", 
-             
-             // The secret key where the Azure Service Bus Topic connection string is located that the background job will use to receive the Azure Key vault events.
-             serviceBusTopicConnectionStringSecretKey: "ARCUS_KEYVAULT_SECRETNEWVERSIONCREATED_CONNECTIONSTRING",
-             
-             // The secret key where the Azure Service Bus connection string is located that your target message pump uses.
-             // This secret key name will be used to check if the received Azure Key Vault event is from this secret or not.
-             messagePumpConnectionStringKey: secretName,
-    
-             // The maximum amount of thrown unauthorized exceptions that your message pump should allow before it should restart either way.
-             // This amount can be used to either wait for an Azure Key Vault event or rely on the thrown unauthorized exceptions.
-             maximumUnauthorizedExceptionsBeforeRestart: 5)
+                .WithServiceBusMessageHandler<OrdersAzureServiceBusMessageHandler, Order>()
+                // This extension will be available to you once you installed the package.
+                .WithAutoRestartOnRotatedCredentials(
+                     jobId: jobId, 
+                     subscriptionNamePrefix: "TestSub", 
+                     
+                     // The secret key where the Azure Service Bus Topic connection string is located that the background job will use to receive the Azure Key vault events.
+                     serviceBusTopicConnectionStringSecretKey: "ARCUS_KEYVAULT_SECRETNEWVERSIONCREATED_CONNECTIONSTRING",
```

Comment:
Is this secret retrieved from configuration or a secret-store ?  And if it's from a secret store, which secret-store ?

Reply:
Secret store, but yeah, is not configured in this sample. We could add it, of course. Thx!

Reply:
Done in https://github.com/arcus-azure/arcus.backgroundjobs/pull/126/commits/be80202742e26bf9c6f2276abfc93f46fed2c42d